utility/common_utils.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file common_utils.cpp
+ * @author Sam Olds (sam@ardusat.com)
+ * @date 4-8-2016
+ * @brief Common I2C read/write functions used by many drivers
+ */
+
+#include "common_utils.h"
+
+/**
+ * Generic I2C Read function
+ *
+ * @param devAddr I2C Device address
+ * @param reg Register address
+ * @param val pointer to byte array to read into
+ * @param length number of bytes to read
+ *
+ * @return 0 on success, other on failure
+ */
+int readFromRegAddr(uint8_t devAddr, uint8_t reg, void *val, int length, endian_e endianness)
+{
+  uint8_t *byteArray = (uint8_t *) val;
+  int ret;
+  int readData = 0;
+
+  Wire.beginTransmission(devAddr);
+
+  if (Wire.write(reg) < 1)
+    return -1;
+
+  if ((ret = Wire.endTransmission(false)) != 0) {
+    return ret;
+  }
+
+  if (byteArray == NULL || length == 0)
+    return 0;
+
+  Wire.beginTransmission(devAddr);
+  Wire.requestFrom((uint8_t) devAddr, (uint8_t) length);
+
+  if (endianness == BIG_ENDIAN) {
+    while (Wire.available() > 0)
+      byteArray[readData++] = Wire.read();
+  } else { // littleEndian
+    while (Wire.available() > 0 && length >= 0)
+      byteArray[--length] = Wire.read();
```

Comment:
This conditional will be executed on every loop... though, I guess this will only loop ~4 times. So I guess there's really not much difference performance-wise.